SimCalorimetry/HGCalSimProducers/test/HGCHEbackSignalScalerAnalyzer.cc

```diff
@@ -0,0 +1,300 @@
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+// Geometry
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+#include "Geometry/HGCalCommonData/interface/HGCalDDDConstants.h"
+#include "Geometry/HcalCommonData/interface/HcalDDDRecConstants.h"
+
+#include "SimCalorimetry/HGCalSimProducers/interface/HGCHEbackDigitizer.h"
+
+//ROOT headers
+#include <TProfile2D.h>
+#include <TH2F.h>
+#include <TF1.h>
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+//STL headers
+#include <vector>
+#include <sstream>
+#include <string>
+#include <iomanip>
+
+#include "vdt/vdtMath.h"
+
+//
+// class declaration
+//
+
+class HGCHEbackSignalScalerAnalyzer : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+public:
+	explicit HGCHEbackSignalScalerAnalyzer(const edm::ParameterSet&);
+	~HGCHEbackSignalScalerAnalyzer() override;
+
+private:
+	void beginJob() override {}
+	void analyze(const edm::Event&, const edm::EventSetup&) override;
+	void endJob() override {}
+
+	void createBinning(const std::vector<DetId>&);
+	void printBoundaries();
+
+	// ----------member data ---------------------------
+	edm::Service<TFileService> fs;
+
+	std::string doseMap_;
+	uint32_t nPEperMIP_;
+
+	std::map<int, std::map<int, float>> layerRadiusMap_;
+	std::map<int, double> layerMap_;
+	std::map<int, std::vector<float>> hgcrocMap_;
+
+	const HGCalGeometry* gHGCal_;
+	const HGCalDDDConstants* hgcCons_;
+
+	int firstLayer_, lastLayer_;
+	float radiusMin_ = 70; //cm
+	float radiusMax_ = 280; //cm
+	int radiusBins_ = 525;
+
+};
+
+//
+// constructors and destructor
+//
+HGCHEbackSignalScalerAnalyzer::HGCHEbackSignalScalerAnalyzer(const edm::ParameterSet& iConfig) :
+doseMap_(iConfig.getParameter<std::string>("doseMap")),
+nPEperMIP_(iConfig.getParameter<uint32_t>("nPEperMIP"))
+{
+	usesResource("TFileService");
+	fs->file().cd();
+}
+
+
+HGCHEbackSignalScalerAnalyzer::~HGCHEbackSignalScalerAnalyzer()
+{
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+void
+HGCHEbackSignalScalerAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+	//get geometry
+	edm::ESHandle<HGCalGeometry> geomhandle;
+	iSetup.get<IdealGeometryRecord>().get("HGCalHEScintillatorSensitive", geomhandle);
+	if(!geomhandle.isValid())
+	{
+		edm::LogError("HGCHEbackSignalScalerAnalyzer") << "Cannot get valid HGCalGeometry Object";
+		return;
+	}
+	gHGCal_ = geomhandle.product();
+	const std::vector<DetId>& detIdVec = gHGCal_->getValidDetIds();
+	std::cout << "total number of cells: " << detIdVec.size() << std::endl;
```

Comment:
This looks like a remnant of some debug line included during development: please remove if so.

Reply:
this is actually useful to keep as it prints the number of cells for the hgc scintillator sector and depending on the version of the geometry the values change significantly:

```
- V9: 376416
- V10: 236160
- V10+PR26504: 225216
```

Reply:
Sorry, I did not notice it is in a /test area
Please forget my comment